test/ln.js

```diff
@@ -96,4 +96,21 @@ assert.equal(
   'new content 3'
 );
 
+// Relative regression
+shell.ln('-sf', 'file1.txt', 'tmp/file2.txt');
+shell.mkdir('-p', 'tmp/new');
+// Move the symlink first, as the reverse confuses `mv`.
+shell.mv('tmp/file2.txt', 'tmp/new/file2.txt');
+shell.mv('tmp/file1.txt', 'tmp/new/file1.txt');
+assert(fs.existsSync('tmp/new/file2.txt'));
+assert.equal(
+  fs.readFileSync('tmp/new/file1.txt').toString(),
+  fs.readFileSync('tmp/new/file2.txt').toString()
+);
+fs.writeFileSync('tmp/new/file1.txt', 'new content txt');
+assert.equal(
+  fs.readFileSync('tmp/new/file2.txt').toString(),
+  'new content txt'
+);
+
```

Comment:
Could you please add a test that uses hard links? the semantics for hard links are a little different.

```
$ mkdir -p tmp/
$ echo 'not in tmp' > ./file1.txt
$ echo 'within tmp' > ./tmp/file1.txt
$ ln -sf file1.txt tmp/softlink.txt
$ ln -f file1.txt tmp/hardlink.txt
$ cat tmp/softlink.txt
within tmp
$ cat tmp/hardlink.txt
not in tmp
```